openeo/extra/job_management/__init__.py

```diff
@@ -668,25 +708,72 @@ def on_job_cancel(self, job: BatchJob, row):
         """
         pass
 
+    def _postprocess_futures(self, worker_pool_results, df, stats):
+        """
+        Processes completed tasks from the worker thread pool and updates job statuses in the job dataframe.
+
+        This method iterates over the results of completed tasks from the worker pool (e.g., job start attempts).
+        For each task, it updates the job's status in the DataFrame based on whether the start was succesful
+        and the job moved to the queued status or wheter the job start failed.
+
+        :param pool_output:
+            A list of tuples, where each tuple contains:
+                - task: An instance of `_JobStartTask` representing the job start task.
+                - result: A tuple containing:
+                    - job_id (str): The ID of the job associated with the task.
+                    - success (bool): Whether the task succeeded.
+                    - data (str): the resulting status or an error message.
+
+        :param df:
+            The dataFrame that specifies the jobs and tracks their statuses. The status of jobs in the DataFrame is updated based on the task results.
+
+        :param stats:
+            The dictionary for tracking statistics related to job processing. The statistics are updated based on the task results.
+
+        """
+
+        for task, result in worker_pool_results:
+            job_id, success, data = result
+
+            # Find rows with matching job id and a status of 'queued_for_start'
+            idx = df.index[(df["id"] == job_id) & (df["status"] == "queued_for_start")]
+            
+            if not idx.empty:
+                # Set new status based on task success
+                new_status = "queued" if success else "start_failed"
+                df.loc[idx, "status"] = new_status
+                _log.info(f"Updated job {job_id} status to {new_status} in dataframe.")
+            else:
+                _log.warning(f"No entry for job {job_id} with status 'queued_for_start' found in dataframe, passing on.")
+            
+            # Log details and update statistics
+            if isinstance(task, _JobStartTask) and success:
+                _log.info(f"Job {job_id} started sucessfully with status: {data}")
```

Comment:
It feels like these logs have a lot of duplication with the logs a couple of lines higher